### PR TITLE
Make "Go To Definition" work on project files

### PR DESCRIPTION
**Pull request by murphm8**
This should work with files that are unsaved, yet still find the right definition.

### CODE REVIEW

File: VisualRust/RustGoToDefinitionCommandHandler.cs [hunk 1]
```diff
@@ -59,7 +59,7 @@ protected override bool Execute(VSConstants.VSStd97CmdID command, uint options,
             }
 
             // TODO: Taken from AugmentCompletionSession
-            using (var tmpFile = new TemporaryFile(snapshot.GetText()))
+            using (var tmpFile = new TemporaryFile(snapshot.GetText(), System.IO.Path.GetDirectoryName(_buffer.GetFilePath())))
```
Comment (vosen) [inline VisualRust/RustGoToDefinitionCommandHandler.cs]: Compiler is returning an error `The name '_buffer' does not exist in the current context`

**Comment by vosen** [comment]
Looks ok, just fix the compiler error

**Comment by LaylConway** [comment]
This code does not clean up after itself, temp files are not automatically deleted on windows. (I recently had Visual Studio crash badly simply because the temp files folder was too full)

**Comment by Boddlnagg** [comment]
@LaylConway: That's why `TemporaryFile` implements `IDisposable`, so at the end of the `using` block, the file should be removed. Or is this somehow not working?

**Comment by LaylConway** [comment]
My bad you're right, skimmed over the code too quickly.

**Comment by murphm8** [comment]
@vosen Finally found some time to dig into this a bit more. I think using DTE to get the currently active path is the right way. I don't think ActiveDocument can be null in this context, since the user is executing "Go To Definition" and the cursor has to be on some text.

**Comment by vosen** [comment]
Yes, using DTE is fine

**Pull request merged**